test/Microsoft.ML.Benchmarks/BenchmarkBase.cs

```diff
@@ -26,11 +28,38 @@ static BenchmarkBase()
         // is running is it sometime cause process hanging when the constructor trying 
         // to load MKL, this is related to below issue:
         // https://github.com/dotnet/machinelearning/issues/1073
-        public static string GetBenchmarkDataPath(string name)
+        public static string GetBenchmarkDataPathAndEncureData(string name, string path = "")
         {
             if (string.IsNullOrWhiteSpace(name))
                 return null;
-            return Path.GetFullPath(Path.Combine(DataDir, name));
+            
+            var filePath = path == "" ? 
+                Path.GetFullPath(Path.Combine(DataDir, name)) : 
+                Path.GetFullPath(Path.Combine(DataDir, path, name));
+
+            if (!File.Exists(filePath))
+            {
```

Comment:
nit: It may be more readable to check if the file exists perform an early return.

Reply:
Got it, thanks

---
In reply to: [423402057](https://github.com/dotnet/machinelearning/pull/5117#discussion_r423402057) [](ancestors = 423402057)